packages/main/src/Menu.ts

```diff
@@ -341,27 +349,45 @@ class Menu extends UI5Element {
 	}
 
 	_itemKeyDown(e: KeyboardEvent) {
-		if (!isLeft(e) && !isRight(e)) {
-			return;
-		}
-
-		const shouldCloseMenu = this.isRtl ? isRight(e) : isLeft(e);
-		const shouldOpenMenu = this.isRtl ? isLeft(e) : isRight(e);
+		const isTabNextPrevious = isTabNext(e) || isTabPrevious(e);
 		const item = e.target as MenuItem;
 		const parentElement = item.parentElement as MenuItem;
+		const shouldCloseMenu = parentElement.hasAttribute("ui5-menu-item") && (this.isRtl ? isRight(e) : isLeft(e));
+		const shouldOpenMenu = this.isRtl ? isLeft(e) : isRight(e);
 
-		if (isEnter(e)) {
-			e.preventDefault();
-		}
-		if (shouldOpenMenu) {
-			this._openItemSubMenu(item);
-		} else if (shouldCloseMenu && parentElement.hasAttribute("ui5-menu-item") && parentElement._popover) {
-			parentElement._popover.open = false;
-			parentElement.selected = false;
-			(parentElement._popover.opener as HTMLElement)?.focus();
+		if (item.hasAttribute("ui5-menu-item")) {
+			if (isEnter(e) || isTabNextPrevious) {
+				e.preventDefault();
+			}
+
+			if (isRight(e) || isLeft(e)) {
+				item._navigateToEndContent(isLeft(e));
+			}
+
+			if (shouldOpenMenu) {
+				this._openItemSubMenu(item);
+			} else if ((shouldCloseMenu || isTabNextPrevious) && parentElement._popover) {
+				parentElement._popover.open = false;
+				parentElement.selected = false;
+				const opener = this._popover.getOpenerHTMLElement(parentElement._popover.opener);
```

Comment:
Move this code to `Popover.ts`, create a new function called `focusOpener()`, add Teo Taushanov and Pepi Dimov for review. 

The new function calls `this.getOpener.......?focus()`, annotate as `@protected`